Codecademy/gamut
# fix(Modal): Give Modal programmatic focus instead of close button

### Overview
Update Modal to give programmatic focus to the Modal instead of the close button. 


Update Modal to give programmatic focus to the Modal instead of the close button. 


### PR Checklist

- [ ] Related to JIRA ticket: [GM-753](https://skillsoftdev.atlassian.net/browse/GM-753)

#### Testing Instructions



1. Go to Gamut's Modal page and "Trigger Modal" (?path=/docs/molecules-modals-modal--trigger-modal) 
2. Turn on VO 
3. Click on "Open Modal" Or "Open multi-view Modal" 
4. Notice that the programmatic focus is on the Modal container (and not the close button) 
5. Tab off the Modal container (into the close button) 
6. Notice that you cannot tab back into the Modal container 
7. Go to Dialog and "Trigger Dialog" (?path=/docs/molecules-modals-dialog--trigger-dialog#triggering-a-dialog) 
8. Click on "Open Dialog"
9. repeat steps 4-6
— 
Mono
1. Use the LE Preview link in the PR!
2. Go to a ex where you can gain progress (e.g. https://tayra.codecademy.com/courses/introduction-to-javascript/lessons/variables/exercises/var?PR_ENV=le-pr-6984)
3. Turn on VO
4. Click "Reset Workspace" at the bottom of the page
5. Notice that the Modal's focus is now on the container 
6. Tab off the Modal container (into the close button) 
7. Notice that you cannot tab back into the Modal container 
...
8. Profit!

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/38034)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/6984)  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |




[GM-753]: https://codecademy.atlassian.net/browse/GM-753?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
